recovery_points_for_virtual_server.rb

```diff
@@ -80,14 +80,20 @@ def get_vm_recovery_points(resource_id)
     vm_points.is_a?(Hash) ? [vm_points] : vm_points
   end
 
-  def get_backup_size(backup_id)
-    api_get(
-      build_query(:BackupFile, BackupUid: identifier_to_uid(:Backup, backup_id))
-    )
-    .dig(:QueryResult, :Entities, :BackupFiles, :BackupFile)
-    .map { |hash| hash.transform_keys(&:downcase) }
-    .max_by { |hash| Time.parse(hash[:creationtimeutc]) if hash[:creationtimeutc] }
-    &.fetch(:backupsize, 0)
+  def get_backup_size(vm_restore_point_hash)
+    refs = vm_restore_point_hash.dig(:Links, :Link)
```

Comment:
vm_restore_point_hash variable here has hash inside like this
`{:Links=>
  {:Link=>
    [{:Href=>"https://10.63.0.92:9398/api/backupServers/bd76362f-fa63-4d9a-a988-599b41c0f423", :Name=>"vc-b-backup.onappdev.lviv", :Type=>"BackupServerReference", :Rel=>"Up"},
     {:Href=>"https://10.63.0.92:9398/api/restorePoints/f1734567-c830-40bd-8476-e935449c788b", :Name=>"Apr  7 2022 11:49AM", :Type=>"RestorePointReference", :Rel=>"Up"},
     {:Href=>"https://10.63.0.92:9398/api/backupFiles/bf1f4635-ead1-4f5a-ae4b-adabac3b0e75",
      :Name=>"Hourly 4@ms_centOS6_vmwaretoolsD2022-04-07T144924_F0CB.vib",
      :Type=>"BackupFileReference",
      :Rel=>"Up"},
     {:Href=>"https://10.63.0.92:9398/api/vmRestorePoints/2e9b79cb-f1ff-4d63-8bf7-c3efdcf0a83c?format=Entity",
      :Name=>"ms_centOS6_vmwaretools-VNAN@2022-04-07 11:49:48",
      :Type=>"VmRestorePoint",
      :Rel=>"Alternate"}]},
 :UID=>"urn:veeam:VmRestorePoint:2e9b79cb-f1ff-4d63-8bf7-c3efdcf0a83c",
 :Name=>"ms_centOS6_vmwaretools-VNAN@2022-04-07 11:49:48",
 :Href=>"https://10.63.0.92:9398/api/vmRestorePoints/2e9b79cb-f1ff-4d63-8bf7-c3efdcf0a83c",
 :Type=>"VmRestorePointReference"}`
 
 It will work